test/e2e/resize_nodes.go

```diff
@@ -560,6 +566,93 @@ var _ = Describe("Nodes [Disruptive]", func() {
+				nodeOpts := api.ListOptions{}
+				nodes, err := c.Nodes().List(nodeOpts)
+				Expect(err).NotTo(HaveOccurred())
+				filterNodes(nodes, func(node api.Node) bool {
+					return isNodeConditionSetAsExpected(&node, api.NodeReady, true)
+				})
+				if len(nodes.Items) <= 0 {
+					Failf("No node were found Ready: %d", len(nodes.Items))
+				}
+				node := nodes.Items[0]
+				podOpts := api.ListOptions{FieldSelector: fields.OneTermEqualSelector(client.PodHost, node.Name)}
+				if err = waitForMatchPodsCondition(c, podOpts, "Running and Ready", podReadyTimeout, podRunningReady); err != nil {
+					Failf("Pods on node %s are not ready and running within %v: %v", node.Name, podReadyTimeout, err)
+				}
+				nodeOpts = api.ListOptions{FieldSelector: fields.OneTermEqualSelector("metadata.name", node.Name)}
+
+				By("Set up watch on node status")
+				w, err := c.Nodes().Watch(nodeOpts)
+				if err != nil {
+					Failf("Failed to set up watch: %v", err)
+				}
+
+				defer w.Stop()
+
+				By(fmt.Sprintf("Performing temporary network partition on node %s", node.Name))
+				partitionCmd := "nohup sh -c 'sleep 10 && sudo ifdown eth0 && sleep 120 && sudo ifup eth0' >/dev/null 2>&1 &"
+				Expect(issueSSHCommand(partitionCmd, testContext.Provider, &node)).NotTo(HaveOccurred())
+
+				By("Expect to observe node and pod status change from Ready to NotReady after network partition")
+				timer := time.After(nodeNotReadyTimeout)
+				notReady := false
+				timeout := false
+				var lastNode *api.Node
+				for !notReady && !timeout {
+					select {
+					case event, _ := <-w.ResultChan():
+						if event.Type == watch.Modified {
+							lastNode = event.Object.(*api.Node)
+							if isNodeConditionSetAsExpected(lastNode, api.NodeReady, false) {
+								notReady = true
+							}
+						}
+					case <-timer:
+						timeout = true
+					}
+				}
+				if !notReady {
+					Fail("Failed to observe node status change from Ready to NotReady")
+				}
+				if err = waitForMatchPodsCondition(c, podOpts, "NotReady", podNotReadyTimeout, podNotReady); err != nil {
+					Failf("Pods on node %s did not become NotReady within %v: %v", node.Name, podNotReadyTimeout, err)
+				}
+
+				By("Expect to observe node and pod status change from NotReady to Ready after network connectivity recovers")
+				timer = time.After(nodeReadyTimeout)
+				ready := false
+				timeout = false
+				for !ready && !timeout {
+					select {
+					case event, _ := <-w.ResultChan():
+						if event.Type == watch.Modified {
+							lastNode = event.Object.(*api.Node)
+							if isNodeConditionSetAsExpected(lastNode, api.NodeReady, true) {
+								ready = true
+							}
+						}
+					case <-timer:
+						timeout = true
+					}
+				}
+				if !ready {
+					Fail("Failed to observe node status change from NotReady to Ready")
+				}
+				if err = waitForMatchPodsCondition(c, podOpts, "Running and Ready", podReadyTimeout, podRunningReady); err != nil {
```

Comment:
The problem with this approach is that it requires to specify namespaces while I want to check all pods on the node. I could not find a util function to list all pods on a node, that is why I wrote a generic one.